ssh_filter_btrbk.sh

```diff
@@ -3,7 +3,7 @@
 set -e
 set -u
 
-export PATH=/sbin:/bin:/usr/sbin:/usr/bin
+export PATH='/usr/bin:/bin'
```

Comment:
Oh and also:<br>
I guess most distros will switch to `/usr/bin`, simply because `btrfs` can be run as a user, too.

And we don't really need to care too much on ancient versions of some distros:<br>
Those would anyway still ship the old `ssh_filter_btrbk.sh`, so people could simply use that.

Reply:
>  I guess most distros will switch to /usr/bin, simply because btrfs can be run as a user, too.

I hope so too, but these things usually take quite a while to propagate through all distros.

Probably the "correct" way would be not to restrict PATH at all, as this should be done in a layer below (i.e. sshd_config). Note that as some exotic distros like NixOS install executables outside `/usr/bin`.

We need to find a good balance between security and convenience here. As long as there are distros out there installing `btrfs` to `/sbin/` (notably Gentoo), we need to keep both.

> If you'd agree with that, we could make the PATH configurable via Makefile, or simply leave it the duty of the distribution maintainer, to override if still using old sbin.

This is asking for trouble. Are you proposing something like "try to find out where third-party software (`btrfs`, ...) was installed, and patch a variable according to it" ?

Reply:
@calestyo Fedora 41, which is one of the biggest distros out there and defaults to btrfs as its FS of choice (thus a high percent of users use btrfs) uses `/usr/sbin` for `btrfs`.

Reply:
Hmm seems a pretty strange choice (I'd call it a bug, TBH), as quite some parts of `btrfs` are usable as normal user.